runtime/bcutil/cfreader.c

```diff
@@ -2103,6 +2134,20 @@ checkAttributes(J9CfrClassFile* classfile, J9CfrAttribute** attributes, U_32 att
 		}
 #endif /* J9VM_OPT_VALHALLA_NESTMATES */
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		case CFR_ATTRIBUTE_ValueTypeObjects: {
+			U_16 valueTypeObjectCount = ((J9CfrAttributeValueTypeObjects*)attrib)->numberOfObjects;
+			for (j = 0; j < valueTypeObjectCount; j++) {
+				value = ((J9CfrAttributeValueTypeObjects*)attrib)->objects[j];
+				if ((0 == value) || (value > cpCount)) {
+					errorCode = J9NLS_CFR_ERR_BAD_INDEX__ID;
+					goto _errorFound;
+				}
+			}
```

Comment:
should also add a check to make sure it is a classref

Reply:
Is there a process for creating the error messages with the translations? Ive put some placeholder name for now but It won't compile since the name doesn't exist in the nls files